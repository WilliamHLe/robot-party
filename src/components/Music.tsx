import React, {useState} from "react";
import "./music/Slider.css";

const Music : React.FC = () => {

    const Music1 = require("./music/Confrontation.mp3");
    const Music2 = require("./music/tragic.mp3");
    const Music3 = require("./music/name.mp3");

    const [selected, setSelected] = useState<string>(Music1);

    const setVolume = () => {
        const musicPlayer:any = document.getElementById("musicPlayer");
        const volumeSlider:any = document.getElementById("volumeSlider");
        console.log(musicPlayer.paused);
        musicPlayer.volume = volumeSlider.value/100;

        if(musicPlayer.paused) {
            musicPlayer.play();
        }
    }

    document.addEventListener("click",function handler(e) {
        const musicPlayer:any = document.getElementById("musicPlayer");
        musicPlayer.play();
        const target:any = e;
        target.currentTarget.removeEventListener(e.type,handler);
    });

    const changeMusic = (newMusic:any) => {
        const musicPlayer:any = document.getElementById("musicPlayer");
        setSelected(newMusic);

        musicPlayer.load();
        musicPlayer.play();
    }

    const pauseMusic = () => {
        const musicPlayer:any = document.getElementById("musicPlayer");
        const pauseButton:any = document.getElementById("pause");
        if(musicPlayer.paused) {
            pauseButton.textContent = "\u23F8\uFE0E";
            musicPlayer.play();
        } else {
            pauseButton.textContent = "\u23F5\uFE0E";
            musicPlayer.pause();
        }
    }

    return (
        <div className="radio-wrapper">
            <div>
                <input type="radio" value="Music1" name="music" checked={selected === Music1} onChange={() => changeMusic(Music1)} /> Musikk 1
                <input type="radio" value="Music2" name="music" checked={selected === Music2} onChange={() => changeMusic(Music2)}/> Musikk 2
                <input type="radio" value="Music3" name="music" checked={selected === Music3} onChange={() => changeMusic(Music3)}/> Musikk 3
                <br />
                <audio id="musicPlayer" onPlay={() => setVolume()} autoPlay>
                    <source src={selected} type="audio/mpeg"/>
                </audio>
                <span id="pause" onClick={pauseMusic}>&#x23F8;&#xFE0E;</span><span className="lowVolume"> </span><input id="volumeSlider" type="range" min="0" max="100" defaultValue="50" className="slider" onChange={() => setVolume()}/><span className="maxVolume"> </span>
            </div>
        </div>
    )
}

export default Music;