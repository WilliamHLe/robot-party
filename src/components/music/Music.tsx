import React, {useState} from "react";
import "./Slider.css";
import "../radio/Radio.css";

const Music : React.FC = () => {

    //Musikk som er brukt
    //Kilder:
    //Moose: https://www.bensound.com/royalty-free-music/track/moose
    //Colour of your face: NoMBe
    //Robots a Cometh: Dan Lebowitz
    const Music1 = require("./music/bensoundmoose.mp3");
    const Music2 = require("./music/ColourOfYourFaceNoMBe.mp3");
    const Music3 = require("./music/RobotsaComethDanLebowitz.mp3");

    const [music, setMusic] = useState<string>(Music1);

    //Funksjon endrer volumet til musikken
    const setVolume = () => {
        const musicPlayer:any = document.getElementById("musicPlayer");
        const volumeSlider:any = document.getElementById("volumeSlider");
        musicPlayer.volume = volumeSlider.value/100;
    }

    //Funksjon som endrer musikken
    //Nødvendig siden audio-taggen må bruke load() for å laste inn ny musikk,
    //og kan derfor ikke bare bruke setMusic()
    const changeMusic = (newMusic:any) => {
        const musicPlayer:any = document.getElementById("musicPlayer");
        setMusic(newMusic);
        musicPlayer.load();
        musicPlayer.play();
        changePButton();
    }

    //Funksjon endrer utseende til pause-knappen basert på musikkens tilstand
    const changePButton =() => {
        const musicPlayer:any = document.getElementById("musicPlayer");
        const pauseButton:any = document.getElementById("pButton");
        if(musicPlayer.paused) {
            pauseButton.className = "";
            pauseButton.className = "pause";
        } else {
            pauseButton.className = "";
            pauseButton.className = "play";
        }
    }

    //Funksjon for å stoppe eller starte musikken
    const pauseMusic = () => {
        const musicPlayer:any = document.getElementById("musicPlayer");
        const pauseButton:any = document.getElementById("pButton");
        (musicPlayer.paused) ? musicPlayer.play() : musicPlayer.pause();
        changePButton();
    }

    return (
        <div className="radio-element">
            <h3>Velg musikk:</h3>
            <div>
                <label>
                    <input type="radio" value={Music1} name="musicChoice1" checked={music === Music1} onChange={() => changeMusic(Music1)} /> Musikk 1
                </label>
                <label>
                    <input type="radio" value={Music2} name="musicChoice2" checked={music === Music2} onChange={() => changeMusic(Music2)}/> Musikk 2
                </label>
                <label>
                    <input type="radio" value={Music3} name="musicChoice3" checked={music === Music3} onChange={() => changeMusic(Music3)}/> Musikk 3
                </label>
                <br />
                <audio id="musicPlayer" onLoad={() => setVolume()} autoPlay>
                    <source src={music} type="audio/mpeg"/>
                </audio>
                {/* Musicspiller */}
                <span id="pButton" className="pause" onLoad={changePButton} onClick={pauseMusic}> </span>
                <span className="lowVolume"> </span>
                {/* Range-input for endring av volum */}
                <input id="volumeSlider" type="range" min="0" max="100" defaultValue="50" className="slider" onChange={() => setVolume()}/>
                <span className="maxVolume"> </span>
            </div>
        </div>
    )
}

export default Music;