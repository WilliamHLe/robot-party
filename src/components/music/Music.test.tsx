import React from 'react';

import Music from "./Music";
const TestRenderer = require('react-test-renderer');

test('renders the music component correctly', () => {
    const component = TestRenderer.create(
        <Music />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});