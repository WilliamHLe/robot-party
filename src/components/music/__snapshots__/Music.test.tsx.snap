// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the music component correctly 1`] = `
<div
  className="radio-element"
>
  <h3>
    Velg musikk:
  </h3>
  <div>
    <label>
      <input
        checked={true}
        name="musicChoice1"
        onChange={[Function]}
        type="radio"
        value="bensoundmoose.mp3"
      />
       Musikk 1
    </label>
    <label>
      <input
        checked={false}
        name="musicChoice2"
        onChange={[Function]}
        type="radio"
        value="ColourOfYourFaceNoMBe.mp3"
      />
       Musikk 2
    </label>
    <label>
      <input
        checked={false}
        name="musicChoice3"
        onChange={[Function]}
        type="radio"
        value="RobotsaComethDanLebowitz.mp3"
      />
       Musikk 3
    </label>
    <br />
    <audio
      autoPlay={true}
      id="musicPlayer"
      onLoad={[Function]}
    >
      <source
        src="bensoundmoose.mp3"
        type="audio/mpeg"
      />
    </audio>
    <span
      className="pause"
      id="pButton"
      onClick={[Function]}
      onLoad={[Function]}
    >
       
    </span>
    <span
      className="lowVolume"
    >
       
    </span>
    <input
      className="slider"
      defaultValue="50"
      id="volumeSlider"
      max="100"
      min="0"
      onChange={[Function]}
      type="range"
    />
    <span
      className="maxVolume"
    >
       
    </span>
  </div>
</div>
`;
